ui/page/transaction/transactions_page.go

```diff
@@ -101,51 +117,31 @@ func (pg *TransactionsPage) OnNavigatedTo() {
 	go pg.scroll.FetchScrollData(false, pg.ParentWindow())
 }
 
-func (pg *TransactionsPage) sectionNavTab(gtx C) D {
-	var selectedTabDims D
-
-	return layout.Inset{
-		Top: values.MarginPadding20,
-	}.Layout(gtx, func(gtx C) D {
-		return pg.tabs.Layout(gtx, len(txTabs), func(gtx C, i int) D {
-			return layout.Stack{Alignment: layout.S}.Layout(gtx,
-				layout.Stacked(func(gtx C) D {
-					return layout.Inset{
-						Right:  values.MarginPadding24,
-						Bottom: values.MarginPadding14,
-					}.Layout(gtx, func(gtx C) D {
-						return layout.Center.Layout(gtx, func(gtx C) D {
-							lbl := pg.Theme.Label(values.TextSize16, txTabs[i])
-							lbl.Color = pg.Theme.Color.GrayText1
-							if i == pg.selectedTabIndex {
-								lbl.Color = pg.Theme.Color.Primary
-								selectedTabDims = lbl.Layout(gtx)
-							}
+// initWalletSelector initializes the wallet selector dropdown to enable
+// filtering transactions for a specific wallet when this page is used to
+// display transactions for multiple wallets.
+func (pg *TransactionsPage) initWalletDropdown() {
+	pg.assetWallets = pg.WL.AllSortedWalletList()
+	if pg.tab.SelectedSegment() != values.String(values.StrTxOverview) {
+		pg.assetWallets = pg.WL.SortedWalletList(utils.DCRWalletAsset)
+	}
 
-							return lbl.Layout(gtx)
-						})
-					})
-				}),
-				layout.Stacked(func(gtx C) D {
-					if i != pg.selectedTabIndex {
-						return D{}
-					}
+	items := []cryptomaterial.DropDownItem{}
+	for _, wal := range pg.assetWallets {
+		assetType := wal.GetAssetType()
+		if assetType == utils.DCRWalletAsset && !pg.hasDCRwallet {
+			pg.hasDCRwallet = true
+		}
 
-					tabHeight := gtx.Dp(values.MarginPadding2)
-					tabRect := image.Rect(0, 0, selectedTabDims.Size.X, tabHeight)
+		item := cryptomaterial.DropDownItem{
+			Text: wal.GetWalletName(),
+			Icon: pg.Theme.AssetIcon(assetType),
+		}
+		items = append(items, item)
+	}
 
-					return layout.Inset{
-						Left: values.MarginPaddingMinus22,
-					}.Layout(gtx, func(gtx C) D {
-						paint.FillShape(gtx.Ops, pg.Theme.Color.Primary, clip.Rect(tabRect).Op())
-						return layout.Dimensions{
-							Size: image.Point{X: selectedTabDims.Size.X, Y: tabHeight},
-						}
-					})
-				}),
-			)
-		})
-	})
+	pg.walletDropDown = pg.Theme.DropDown(items, values.WalletsDropdownGroup, 0)
+	pg.selectedWallet = pg.assetWallets[pg.walletDropDown.SelectedIndex()]
```

Comment:
I just remembered something VERY important. There should not be a default selection. By default, txs for all wallets should be displayed, until and unless a user manually selects a wallet from the wallet dropdown.

This means that `pg.selectedWallet` could be nil, so always check for that before trying to use it.

Reply:
Ermmmmm i did this earlier and the comment was to ensure it was per wallet. No sure i am going back to implementing it.

Reply:
Also, to load txs, we need a wallet and currently, we can are only loading 20txs at a time. If this will be implemented, it will be in a totally different PR.

Reply:
> Ermmmmm i did this earlier and the comment was to ensure it was per wallet. No sure i am going back to implementing it.

reference the comment please.

if iirc correctly what was initially implemented was an asset dropdown and now a wallet dropdown e.g
{btc asset}
{dcr asset}

clicking on btc assets showed you transactions for all your btc wallets combined

i left a comment saying that was not correct it should be a wallet dropdown, e.g all wallets you own would show on the dropdown, so that when a wallet is selected it shows transactions for that wallet only.

>There should not be a default selection

@itswisdomagain comment is correct, there should not be a default wallet being selected on initial page visit, transactions for all wallets you own should show up, until you specifically select a wallet from the dropdown and only then do you see tx for the particular wallet selected